pkg/asset/installconfig/installconfig.go

```diff
@@ -157,6 +157,13 @@ func (a *InstallConfig) finish(filename string) error {
 	}
 	if a.Config.Azure != nil {
 		a.Azure = icazure.NewMetadata(a.Config.Azure.CloudName, a.Config.Azure.ARMEndpoint)
+		if a.Config.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet != nil && a.Config.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.SubscriptionID == "" {
+			session, err := a.Azure.Session()
+			if err != nil {
+				return errors.Wrap(err, "failed to fetch session")
+			}
+			a.Config.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.SubscriptionID = session.Credentials.SubscriptionID
+		}
```

Comment:
So we are loading the value correctly when using the DiskEncryptionSet [for master](https://github.com/openshift/installer/blob/master/pkg/asset/machines/master.go#L354-L357) and [for worker](https://github.com/openshift/installer/blob/master/pkg/asset/machines/worker.go#L439-L442) but the problem was not doing that for the validation?

Reply:
Yes.